src/wrappers/msvc_wrapper.cpp

```diff
@@ -214,13 +214,19 @@ std::map<std::string, std::string> msvc_wrapper_t::get_relevant_env_vars() {
 }
 
 std::string msvc_wrapper_t::get_program_id() {
+  const auto& pathToCL = m_args[0];
+  const auto vsFolderPos = pathToCL.rfind("Microsoft Visual Studio");
```

Comment:
This will give different hashes for VS 2019 Community and Professional, true. But on the other hand this will work with `Microsoft Visual Studio 14` and the like (different Visual Studio versions install into different directories)